code/modules/cooking/machinery/cooking_machines/stove.dm

```diff
@@ -0,0 +1,43 @@
+/obj/machinery/appliance/cooker/stove
+	name = "stove"
+	desc = "Don't touch it!"
+	icon_state = "stove_off"
+	cook_type = "pan-fried"
+	appliancetype = SKILLET | SAUCEPAN | POT
+	food_color = "#a34719"
+	can_burn_food = TRUE
+	active_power_usage = 6 KILOWATTS
+	heating_power = 6000
+	on_icon = "stove_on"
+	off_icon = "stove_off"
+
+	resistance = 5000 // Approx. 2 minutes.
+	idle_power_usage = 1 KILOWATTS
+	//uses ~30% power to stay warm
+	optimal_temp = T0C + 100 // can boil water!
+	optimal_power = 1.2
+
+	max_contents = 4
+
+	stat = POWEROFF	//Starts turned off
+
+/obj/machinery/appliance/cooker/stove/update_icon()
+	. = ..()
+	cut_overlays()
+	var/list/pans = list()
+	for(var/obj/item/reagent_containers/cooking_container/CC in contents)
+		var/image/pan_overlay
+		switch(CC.appliancetype)
+			if(SKILLET)
+				pan_overlay = image('icons/obj/cooking_machines.dmi', "skillet[Clamp(length(pans)+1, 1, 4)]")
+			if(SAUCEPAN)
+				pan_overlay = image('icons/obj/cooking_machines.dmi', "pan[Clamp(length(pans)+1, 1, 4)]")
+			if(POT)
+				pan_overlay = image('icons/obj/cooking_machines.dmi', "pot[Clamp(length(pans)+1, 1, 4)]")
```

Comment:
I feel like these can be cached when used...?

Reply:
Ooor maybe not actually since these can change at any time, probably not worth it

Reply:
There's 16 total. It miiiiiight be needed? But while it does change pretty often, it uses overlays, not icon ops, so it's not like it's sending them to clients every time it's changed.